ui/views/widget/desktop_aura/desktop_drag_drop_client_ozone.h

```diff
@@ -0,0 +1,168 @@
+#include "ui/aura/client/drag_drop_client.h"
+#include "ui/aura/client/drag_drop_delegate.h"
+#include "ui/aura/window_observer.h"
+#include "ui/base/dragdrop/os_exchange_data.h"
+#include "ui/gfx/geometry/point.h"
+#include "ui/views/views_export.h"
+
+namespace ui {
+
+class OSExchangeDataProviderAura;
+class PlatformWindow;
+
+}  // namespace ui
+
+namespace views {
+
+class VIEWS_EXPORT DesktopDragDropClientOzone
+    : public aura::client::DragDropClient,
+      public aura::WindowObserver {
+ public:
+  explicit DesktopDragDropClientOzone(
+      aura::Window* root_window,
+      views::DesktopNativeCursorManager* cursor_manager,
+      ui::PlatformWindow* platform_window);
+  ~DesktopDragDropClientOzone() override;
+
+  // Overridden from aura::client::DragDropClient:
+  int StartDragAndDrop(const ui::OSExchangeData& data,
+                       aura::Window* root_window,
+                       aura::Window* source_window,
+                       const gfx::Point& root_location,
+                       int operation,
+                       ui::DragDropTypes::DragEventSource source) override;
+  void DragCancel() override;
+  bool IsDragDropInProgress() override;
+  void AddObserver(aura::client::DragDropClientObserver* observer) override;
+  void RemoveObserver(aura::client::DragDropClientObserver* observer) override;
+
+  // Overridden from void aura::WindowObserver:
+  void OnWindowDestroying(aura::Window* window) override;
+
+  void OnDragEnter(ui::PlatformWindow* window,
+                   float x,
+                   float y,
+                   const std::vector<std::string>& mime_types,
+                   uint32_t serial,
+                   int operation);
+  void OnDragMotion(float x, float y, uint32_t time, int operation);
+  void OnDragDrop(base::OnceClosure task);
+  void OnDragDataReceived(const std::string& data);
+  void OnDragLeave();
+  void OnDragDataRequest(const std::string& mime_type, std::string* buffer);
+  void OnDragSessionClose();
+  void PerformDrop();
+
+ private:
+  class DragDataCollector {
+   public:
+    DragDataCollector(
+        base::WeakPtr<DesktopDragDropClientOzone> drag_drop_client,
+        const std::vector<std::string>& mime_types);
+    ~DragDataCollector();
+    void HandleNextMimeType();
+
+    ui::OSExchangeData* GetData() const { return os_exchange_data_.get(); }
+    void SetData(std::unique_ptr<ui::OSExchangeData> data) {
+      os_exchange_data_ = std::move(data);
+    }
+    std::list<std::string>& GetUnProcessedMimeTypes() {
+      return unprocessed_mime_types_;
+    }
+
+   private:
+    // Returns the next MIME type to be received from the source process, or an
+    // empty string if there are no more interesting MIME types left to process.
+    std::string SelectNextMimeType();
+
+    base::WeakPtr<DesktopDragDropClientOzone> drag_drop_client_;
+    std::unique_ptr<ui::OSExchangeData> os_exchange_data_;
+    std::list<std::string> unprocessed_mime_types_;
+  };
+
+ public:
```

Comment:
AFAIK, chromium style disallows having 
public:
private:
public:

Can you use forward declaration instead and move the class above into the private section bellow?